support-frontend/assets/helpers/subscriptionsForms/subscriptionCheckoutReducer.ts

```diff
@@ -91,84 +83,14 @@ function createReducer(
 			marketingConsent: marketingConsentReducer,
 			csrf: csrfReducer,
 			recaptcha: recaptchaReducer,
+			deliveryAddress: deliveryAddressReducer,
+			billingAddress: billingAddressReducer,
 		}),
 		user: createUserReducer(),
 		directDebit,
-		...addressReducers,
 	});
 }
 
-function createCheckoutReducer(
-	initialCountry: IsoCountry,
-	product: SubscriptionProduct,
-	initialBillingPeriod: BillingPeriod,
-	startDate: Option<string>,
-	productOption: Option<ProductOptions>,
-	fulfilmentOption: Option<FulfilmentOptions>,
-) {
-	const address = {
-		billingAddress: addressReducerFor('billing', initialCountry),
-	};
-	return createReducer(
-		product,
-		initialBillingPeriod,
-		startDate,
-		productOption,
-		fulfilmentOption,
-		address,
-	);
-}
-
-function createWithDeliveryCheckoutReducer(
-	initialCountry: IsoCountry,
-	product: SubscriptionProduct,
-	initialBillingPeriod: BillingPeriod,
-	startDate: Option<string>,
-	productOption: Option<ProductOptions>,
-	fulfilmentOption: Option<FulfilmentOptions>,
-) {
-	const addresses = {
-		billingAddress: addressReducerFor('billing', initialCountry),
-		deliveryAddress: addressReducerFor('delivery', initialCountry),
-	};
-	return createReducer(
-		product,
-		initialBillingPeriod,
-		startDate,
-		productOption,
-		fulfilmentOption,
-		addresses,
-	);
-}
```

Comment:
I've removed the two separate reducers and just left a single one. This means we'll always have a delivery address even if we don't need it, but I think it simplifies the code

Reply:
👏 Yes, I love this! It's going to be so much easier to handle a more consistent state shape that we just don't use certain bits of in some checkouts.